integreat_cms/cms/migrations/0122_add_archive_contact_permission.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from django.db import migrations
+
+from ..constants import roles
+
+if TYPE_CHECKING:
+    from django.apps.registry import Apps
+    from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+
+def update_roles(
+    apps: Apps,
+    _schema_editor: BaseDatabaseSchemaEditor,
+) -> None:
+    """
+    Update the permission settings for the model contact
+
+    :param apps: The configuration of installed applications
+    """
+    # We can't import the Person model directly as it may be a newer
+    # version than this migration expects. We use the historical version.
+    Group = apps.get_model("auth", "Group")
+    Permission = apps.get_model("auth", "Permission")
+
+    # Assign the correct permissions
+    for role_name in dict(roles.CHOICES):
+        group, _ = Group.objects.get_or_create(name=role_name)
+        # Clear permissions
+        group.permissions.clear()
+        # Set permissions
+        group.permissions.add(
+            *Permission.objects.filter(codename__in=roles.PERMISSIONS[role_name]),
+        )
+
+
+class Migration(migrations.Migration):
+    """
+    Migration file to update contact permissions
+    """
+
+    dependencies = [
+        ("cms", "0121_remove_userchat_most_recent_hits_and_more"),
+    ]
```

Comment:
@jonbulz 

Please update the numbering and order of migration files as #3552 is merged with a new migration file 🙏 🥺 
Otherwise we would have two files as children of `0121_remove_userchat_most_recent_hits_and_more`.